lib/ice_cube/parsers/ical_parser.rb

```diff
@@ -1,83 +1,102 @@
-              occ = expr[0..-3].to_i
-              dows[day].nil? ? dows[day] = [occ] : dows[day].push(occ)
-              days.delete(TimeUtil.sym_to_wday(day))
-            else
-              days.push TimeUtil.sym_to_wday(day) if dows[day].nil?
+        ical.split(';').each do |rule|
+          (name, value) = rule.split('=')
+          raise ArgumentError, "Invalid iCal rule component" if value.nil?
+          value.strip!
+          case name
+          when 'FREQ'
+            params[:rule_type] = "IceCube::#{value[0]}#{value.downcase[1..-1]}Rule"
+          when 'INTERVAL'
+            params[:interval] = value.to_i
+          when 'COUNT'
+            params[:count] = value.to_i
+          when 'UNTIL'
+            params[:until] = Time.parse(value).utc
+          when 'WKST'
+            params[:week_start] = TimeUtil.ical_day_to_symbol(value)
+          when 'BYSECOND'
+            validations[:second_of_minute] = value.split(',').map(&:to_i)
+          when 'BYMINUTE'
+            validations[:minute_of_hour] = value.split(',').map(&:to_i)
+          when 'BYHOUR'
+            validations[:hour_of_day] = value.split(',').map(&:to_i)
+          when 'BYDAY'
+            dows = {}
+            days = []
+            value.split(',').each do |expr|
+              day = TimeUtil.ical_day_to_symbol(expr.strip[-2..-1])
+              if expr.strip.length > 2  # day with occurence
+                occ = expr[0..-3].to_i
+                dows[day].nil? ? dows[day] = [occ] : dows[day].push(occ)
+                days.delete(TimeUtil.sym_to_wday(day))
+              else
+                days.push TimeUtil.sym_to_wday(day) if dows[day].nil?
+              end
             end
+            validations[:day_of_week] = dows unless dows.empty?
+            validations[:day] = days unless days.empty?
+          when 'BYMONTHDAY'
+            validations[:day_of_month] = value.split(',').map(&:to_i)
+          when 'BYMONTH'
+            validations[:month_of_year] = value.split(',').map(&:to_i)
+          when 'BYYEARDAY'
+            validations[:day_of_year] = value.split(',').map(&:to_i)
+          when 'BYSETPOS'
+          else
+            validations[name] = nil # invalid type
           end
-          validations[:day_of_week] = dows unless dows.empty?
-          validations[:day] = days unless days.empty?
-        when 'BYMONTHDAY'
-          validations[:day_of_month] = value.split(',').map(&:to_i)
-        when 'BYMONTH'
-          validations[:month_of_year] = value.split(',').map(&:to_i)
-        when 'BYYEARDAY'
-          validations[:day_of_year] = value.split(',').map(&:to_i)
-        when 'BYSETPOS'
-        else
-          validations[name] = nil # invalid type
         end
+
+        Rule.from_hash(params)
       end
 
-      Rule.from_hash(params)
+      private
+
+      def parse_in_tzid(value, tzid)
+        time_parser = Time
+        if tzid
+          time_parser = ActiveSupport::TimeZone.new(tzid.split('=')[1]) || Time
```

Comment:
Use `rescue Time` here if ActiveSupport is not loaded? Maybe something like `IcalParser.time_parser` can be determined statically for performance.